app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -95,6 +136,72 @@ public void afterTextChanged(final Editable s) {
         setEmojiCompatEnabled(false);
     }
 
+    @Override
+    protected void onFinishInflate() {
+        super.onFinishInflate();
+        _defaultPaddingLeft = getPaddingLeft();
+    }
+
+    @Override
+    public boolean onPreDraw() {
+        _paint.setTextSize(getTextSize());
+        _paint.setTextAlign(Paint.Align.RIGHT);
+        _scrollView = getParent() instanceof ScrollView ? (ScrollView) getParent() : (ScrollView) getParent().getParent();
+        return super.onPreDraw();
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+        // If line numbers can be drawn
+        if (_nuEnabled && _maxLineNumber < (AppSettings._isDeviceGoodHardware ? 5000 : 3000)) {
```

Comment:
As line numbers have to be enabled manually, perhaps we should just issue a toast warning instead of hard blocking on `isDeviceGoodHardware`

Reply:
We should also just try to make line numbers dynamic (draw them only for visible lines). That will avoid this completely. Lets remove this test for the line count at first and then I can help with the dynamic line numbers...